### ISSUE TITLE
newPool fills pool.ch with 10000 buffers, but it is never used

### ISSUE THREAD

**Issue by ekle** [shiyanhui/hero#3]

**Comment by shiyanhui** [comment]
Yes, you are right. I did it on purpose. Cause it's much faster with it than without it.

**Comment by ekle** [comment]
would it make sense to put them in the pool ?

**Comment by shiyanhui** [comment]
Nope, because the gc will clear them if they are in sync.Pool. I test it in go1.7 on mac and it did run faster.

**Comment by funny-falcon** [comment]
So that, sync.Pool guarantees that memory is not overused.
I doubt one allocation per GC really matters that much. Probably I'm mistaken.

> I test it in go1.7 on mac and it did run faster.

Was it benchmark or realword workload ?
